client/components/tinymce/plugins/simple-payments/dialog/index.jsx

```diff
@@ -419,11 +420,18 @@ class SimplePaymentsDialog extends Component {
 		);
 	};
 
+	// The only thing we track about image in Simple Payments product form is its media id.
+	// However this doesn't change when the image is edited, and as a result redux form
+	// isDirty selector is not detecting any update, so it's not possible to submit changes
+	// after editing the image (even though they are saved behind the scenes in Media library).
+	// This allows us to force re-enabling of the save button in that case.
```

Comment:
If this is the case then after an edit, the image is updated automatically for payment buttons, correct? 

I wonder if it would be better to update the messaging in the image editor versus marking the product as dirty, since clicking on `done` or `cancel` won't have any effect in this case.

Reply:
For example, we might update the primary copy to say "Update Payment Button" instead of "Done"

<img width="276" alt="screen shot 2018-05-15 at 1 49 07 pm" src="https://user-images.githubusercontent.com/1270189/40082778-da9fb86c-5846-11e8-8699-0ed4329e685e.png">

Reply:
> If this is the case after an edit, the image is updated automatically for payment buttons, correct?

Yes, even before the form is saved, if that was the question.

> I wonder if it would be better to update the messaging in the image editor versus marking the product as dirty, since clicking on done or cancel won't have any effect in this case.

Good point, that might make more sense in this context. :thinking: 

> I'm pretty on the fence here since done, doesn't have any effect here.

Not directly related to Simple Payments work, but if the image editor created a separate copy with new media id, I think a lot of the issues that Kirk mentioned would be resolved. Editing an image in place and preserving the same id can have a lot of other undesired consequences in other areas too (for example reusing the same image but slightly edited in the same post, or wanting to reuse part of the old image by cropping when that image was already added to some other posts). Not sure what were the reasons behind implementing it this way, but it seems like a very strange decision.
However, I suspect that changing that behavior would have a lot of consequences for existing functionality and would require more work and testing.

Reply:
> ... if the image editor created a separate copy with new media id, I think a lot of the issues that Kirk mentioned would be resolved... However, I suspect that changing that behavior would have a lot of consequences for existing functionality and would require more work and testing.

Great idea, and couldn't agree more 🙃

Reply:
>  if the image editor created a separate copy with new media id

That was the original proposed implementation, and it wasn't ideal, as folks also didn't want a bunch of copies cluttering the folder. I'm also certain that I'm forgetting a number of corner cases that would need to be addressed. At the time I believe we decided to match functionality in the core image editor, which has similar behavior to the Calypso implementation.

https://github.com/Automattic/wp-calypso/pull/7446
https://github.com/Automattic/wp-calypso/issues/8081

Maybe @retrofox remembers more?

Reply:
We can certainly revisit the issue in the future, though I do think we'd need to address the paid storage issue if we chose to move to copies, or allow proper image versioning.